### PR TITLE
refactor: v1 recordbatch to json

### PR DESCRIPTION
**Pull request by JeanArhancet**
This PR addresses the second part of issue https://github.com/influxdata/influxdb/issues/24981. It replaces the usage of `&arrow_json::writer::record_batches_to_json_rows(batches.as_slice())?` in `influxdb3_server/src/http/v1.rs`  by directly parsing `RecordBatch` into `serde_json::Value`.

Check-list:
- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: influxdb3_server/src/http/v1.rs [hunk 1]
```diff
@@ -445,6 +454,113 @@ fn convert_ns_epoch(value: Value, precision: Precision) -> Result<Value, anyhow:
...
+                column
+                    .as_primitive::<Float16Type>()
+                    .value(row_index)
+                    .to_f64(),
+            )
+            .context("failed to downcast Float16 column")?,
+        ),
+        DataType::Float32 => Value::Number(
+            serde_json::Number::from_f64(
+                column.as_primitive::<Float32Type>().value(row_index) as f64
```
Comment (hiltontj) [inline influxdb3_server/src/http/v1.rs:498]: You should be able to use `.into()` instead of `as f64` here.

**Review by hiltontj — CHANGES_REQUESTED** [review]
Hey @JeanArhancet - I have a few requested changes in-line.

In addition to those, the existing tests for this API are passing, which is awesome, but since you are adding conversion logic in this PR, I think it would be good to test out the different data types that we can write in using line protocol. The current test only writes in float fields here: https://github.com/influxdata/influxdb/blob/b6718e59e3f0fee50c0bbb2492c7a24cd4d48394/influxdb3/tests/server/query.rs#L678-L693

I think it would be good enough to add a unit test alongside the conversion function you wrote to test out the data-type conversions we are interested in; namely: boolean, string (`Utf8`), integer (`Int64`), and unsigned integer (`UInt64`)

You could also test this from the end-to-end test suite I linked above by writing the values in using line protocol, where you could include some nulls, to make sure the conversions are working end-to-end.

**Comment by JeanArhancet** [comment]
@hiltontj Thanks for your feedback. I have a question concerning the Timestamp conversion. I'm not really confident in my current implementation and unsure if the format is correct. Does Influx v1 support RFC 3339 with the 'Z' suffix for UTC? In the tests, the datetime format used is "%Y-%m-%dT%H:%M:%S".

**Comment by hiltontj** [comment]
> Does Influx v1 support RFC 3339 with the 'Z' suffix for UTC?

@JeanArhancet - that is a good question thanks for raising it. I looked at our documentation, which indicates that it uses RFC 3339 with `Z` (see response examples [here](https://docs.influxdata.com/influxdb/v1/tools/api/#query-data-with-a-select-statement-and-return-pretty-printed-json)). If the values coming out of the DB have timestamps, then we should be  expressing them as UTC along with the `Z` (and in which case, the tests we have will need to be fixed).

If however, the timestamps are not holding a timestamp that may be a separate issue, and you could leave yours as is, since it is passing the tests.

Interestingly, I also tried the `/query` API on my local v1 OSS influxdb and it returns the time as a unix epoch (nanoseconds), not an RFC timestamp. Either way, we want to be consistent with the docs on this one.

File: influxdb3_server/src/http/v1.rs (hunk 1 above)
Comment (JeanArhancet) [reply influxdb3_server/src/http/v1.rs]: Fixed

File: influxdb3_server/src/http/v1.rs [hunk 2]
```diff
@@ -337,41 +341,46 @@ impl QueryResponseStream {
...
+        let schema = batch.schema();
+
+        for row_index in 0..batch.num_rows() {
+            let mut row = vec![Value::Null; column_map.len()];
+
+            for (col_index, column) in columns.iter().enumerate() {
+                let field = schema.field(col_index);
+                let column_name = field.name();
+
+                let mut cell_value = if !column.is_valid(row_index) {
```
[the remark this answers is not in the document]
Comment (JeanArhancet) [reply influxdb3_server/src/http/v1.rs:489]: Fixed

**Comment by JeanArhancet** [comment]
> > Does Influx v1 support RFC 3339 with the 'Z' suffix for UTC?
> 
> @JeanArhancet - that is a good question thanks for raising it. I looked at our documentation, which indicates that it uses RFC 3339 with `Z` (see response examples [here](https://docs.influxdata.com/influxdb/v1/tools/api/#query-data-with-a-select-statement-and-return-pretty-printed-json)). If the values coming out of the DB have timestamps, then we should be expressing them as UTC along with the `Z` (and in which case, the tests we have will need to be fixed).
> 
> If however, the timestamps are not holding a timestamp that may be a separate issue, and you could leave yours as is, since it is passing the tests.
> 
> Interestingly, I also tried the `/query` API on my local v1 OSS influxdb and it returns the time as a unix epoch (nanoseconds), not an RFC timestamp. Either way, we want to be consistent with the docs on this one.

@hiltontj  I've changed the datetime conversion to return an RFC 3339 compliant date. Locally, I also tried this metric, but I encountered an issue:

```rust
 server
        .write_lp_to_db(
            "foo",
          "errors,message='Hello world' count=-9223372036854775808 1422568543702900257\n\

            Precision::Nanosecond,
        )
        .await
        .unwrap();
```

And I've this errors:
```
called `Result::unwrap()` on an `Err` value: ApiError { code: 400, message: "{\"error\":\"partial write of line protocol occurred\",\"data\":[{\"original_line\":\"errors,message='hello world' count=-9223372036854775808 1422568543702900257\",\"line_number\":1,\"error_message\":\"No fields were provided\"}]}" } 
```
However, according to the documentation for v1, this seems to be a supported format: https://docs.influxdata.com/influxdb/v1/write_protocols/line_protocol_reference/#write-the-field-value-stringing-along-as-a-string-to-influxdb?

Additionally, the i64 values are currently being converted to scientific notation, e.g., -9223372036854775808 -> -9.223372036854776e18. I'm not sure if this respects the v1 specification.

**Comment by JeanArhancet** [comment]
> @JeanArhancet I suspect the parser is having trouble with the tag value you provided for `message`, i.e., `'Hello World'`.
> 
> 1. Generally, an error or log message would be better in a field, not a tag
> 2. If you do want to have it in a tag, I would try passing it like this:
> 
> ```
> errors,message=Hello\ World count=...
> ```
> 
> i.e., remove the `'`s, and escape the space between `Hello` and `World`. See [here](https://docs.influxdata.com/influxdb/v1/write_protocols/line_protocol_tutorial/#special-characters).

Thanks for your feedback, @hiltontj. I've updated the test to match the values in this documentation: https://docs.influxdata.com/influxdb/v1/write_protocols/line_protocol_tutorial/#data-types. What do you think?

**Review by hiltontj — APPROVED** [review]
@JeanArhancet I think this is looking really solid - and it is coming at the perfect time, given that the deprecated function you are replacing was just removed in `v52.0.0` of `arrow-json` 😅.

**Pull request merged**